src/controllers/hid/hidcontroller.cpp

```diff
@@ -194,12 +195,23 @@ QByteArray HidController::getInputReport(unsigned int reportID) {
 
 bool HidController::poll() {
     Trace hidRead("HidController poll");
-
-    // This loop risks becoming a high priority endless loop in case processing
-    // the mapping JS code takes longer than the controller polling rate.
-    // This could stall other low priority tasks.
-    // There is no safety net for this because it has not been demonstrated to be
-    // a problem in practice.
+    // This function reads the available HID Input Reports using hidapi
+    // Important to know is, that this reading is not a hardware operation,
+    // instead it reads previously received HID Input Reports from a ring buffer.
+    // Depending on the hidapi backend implementation the ring buffer is either part
+    // of the hidapi implementation or the OS kernel.
+    // The size of the ring buffer also depends on the hidapi backend:
+    // - hidraw(2048 bytes) - Linux Kernel API
+    // - libusb(30 reports) - BSD (alternative Linux userspace implementation)
+    // - mac(30 reports)
+    // - windows(64 reports)
+    // Behavior:
+    // - If the polling rate is to slow, multiple HID Input Reports will be processed at the same time
+    // - If the mapping processes the Input Reports too slow, whole Input Reports will be omitted if the ring buffer is full
+    // - If the processing of all received HID Input Reports in the JS mapping code takes longer than the interval between incoming reports,
+    //   it could stall other low priority tasks.
+
+    int inputReportCounter[kMaxHidInputReportId + 1] = {0};
```

Comment:
```suggestion
    using arr_elem_t = std::remove_all_extents<decltype(m_pPollData)>::type;
    constexpr int kMaxHidInputReportId = std::numeric_limits<arr_elem_t>::max();
    std::array<int, kMaxHidInputReportId> inputReportCounter {};
```
requires
```c++
#include <type_traits>
#include <limits>
#include <array>
```
then remove the kMaxHidInputReportId from the anonymous namespace. 

This is quite some C++ introspection foo. I like it, because it adapts when the type of `m_pPollData` data changes and it avoids a global, but other C++ people might disagree. Thoughts?

Reply:
I think this is not the same, because m_pPollData has always 2 elements:
https://github.com/mixxxdj/mixxx/blob/24c26c79796da4a86318c54dc4c65cfe2b2dd2f2/src/controllers/hid/hidcontroller.h#L74-L76
The number of poll buffers is currently not related to the number of input reports.

Reply:
`std::remove_all_extents<decltype(m_pPollData)>::type` makes sure to remove all extras from the type. `arr_elem_t` is basically `unsigned char`

Reply:
Calling `std::numeric_limits<>` on an array type doesn't even compile.

Reply:
Now I got it! Thanks for the explanation! With the latest commit, this constant is no longer needed at all.